storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -391,6 +391,8 @@ private void initializeResources() {
         boolean topicCreated = false;
         long startTimeMs = time.milliseconds();
         Admin adminClient = null;
+
+        lock.writeLock().lock();
```

Comment:
IIRC, the reason for not taking `writeLock` before the creating of the Admin client is that the topic exists/creation operation is costly when the controller queue is busy. If we take a lock, then the callers of the RLMM might be waiting for this operation to complete.

Reply:
yes, understood